Formula/lilypond.rb

```diff
@@ -0,0 +1,98 @@
+class Lilypond < Formula
+  desc "Music notation for everyone"
+  homepage "https://lilypond.org"
+  url "https://lilypond.org/download/sources/v2.22/lilypond-2.22.1.tar.gz"
+  sha256 "72ac2d54c310c3141c0b782d4e0bef9002d5519cf46632759b1f03ef6969cc30"
+  license "GPL-3.0-or-later"
+
+  depends_on "autoconf" => :build
+  depends_on "fontforge" => :build
+  depends_on "gettext" => :build
+  depends_on "glib" => :build
+  depends_on "imagemagick" => :build
+  depends_on "pkg-config" => :build
+  depends_on "t1utils" => :build
+  depends_on "texi2html" => :build
+
+  depends_on "fontconfig"
+  depends_on "freetype"
+  depends_on "ghostscript"
+  depends_on "guile@2"
+  depends_on :macos
+  depends_on "pango"
+  depends_on "python@3.9"
+
+  uses_from_macos "flex" => :build
+  uses_from_macos "m4" => :build
+
+  resource "bison" do
+    url "https://ftp.gnu.org/gnu/bison/bison-3.7.6.tar.xz"
+    sha256 "67d68ce1e22192050525643fc0a7a22297576682bef6a5c51446903f5aeef3cf"
+  end
+
+  resource "urw-fonts" do
+    url "https://github.com/jsfelix/lilypond-build-deps/releases/download/v2.22.1/urw-base35-fonts-20200910.tar.gz"
+    sha256 "e0d9b7f11885fdfdc4987f06b2aa0565ad2a4af52b22e5ebf79e1a98abd0ae2f"
+  end
+
+  resource "texinfo" do
+    url "https://ftp.gnu.org/gnu/texinfo/texinfo-6.8.tar.xz"
+    sha256 "8eb753ed28bca21f8f56c1a180362aed789229bd62fff58bf8368e9beb59fec4"
+  end
+
+  resource "tex-build-dep" do
+    url "https://github.com/jsfelix/lilypond-build-deps/releases/download/v2.22.1/tex-build-dep-20210909.tar.gz"
+    sha256 "18132bb70b109c6d0399095bc0acaf2616bf9edbd401af4a966eb3807b47797a"
+  end
```

Comment:
What are these build dependencies? These should be built from source. The only exception is when Lilypond needs itself to bootstrap.

Reply:
There is some binary files provided by MacTex or TexLive, used only for build.

Reply:
I don't know exactly how it works, but lilypond needs bibtex, latex, mf, metapost and other tools from the TeX package (provided by MacTeX or Tex-Live), maybe to compile the documentation.

Reply:
According to the docs (https://lilypond.org/doc/v2.23/Documentation/contributor/requirements-for-compiling-lilypond#other), the following TeX dependencies are required to compile Lilypond:
* MetaFont: The MetaFont binary (usually called mf-nowin, mf, mfw, or mfont) and its support files are normally packaged along with TeX. Most GNU/Linux and other free software distributions already provide packages for TeX Live, see above. TeX Live can can also be installed separately; it contains stand-alone binaries for most platforms.
* MetaPost: The mpost binary is also usually packaged with TeX.
* TeX Gyre font packages in OTF format. Also see Fonts

Reply:
Since this formula seems to be partially based on

https://github.com/nwhetsell/homebrew-lilypond/blob/63f3be3914d54d19fc86efa234a56019f5727a86/Formula/lilypond.rb#L30-L127

I thought I should note that, over there, I’m handling the dependency on TeX Live as a [requirement on MacTeX](https://github.com/nwhetsell/homebrew-lilypond/blob/63f3be3914d54d19fc86efa234a56019f5727a86/Library/MacTeXRequirement.rb), although I don’t believe this would be suitable for homebrew-core. It seems like https://github.com/Homebrew/homebrew-core/pull/83738 would need to be merged before LilyPond can be included in homebrew-core.

Reply:
I should also note that according to LilyPond’s documentation, using Guile v2 with LilyPond is not officially supported:

https://lilypond.org/doc/v2.22/Documentation/contributor/requirements-for-running-lilypond

Reply:
You should make formula for these dependencies if you need them.

Reply:
> Since this formula seems to be partially based on
> 
> https://github.com/nwhetsell/homebrew-lilypond/blob/63f3be3914d54d19fc86efa234a56019f5727a86/Formula/lilypond.rb#L30-L127
> 
> I thought I should note that, over there, I’m handling the dependency on TeX Live as a [requirement of MacTeX](https://github.com/nwhetsell/homebrew-lilypond/blob/63f3be3914d54d19fc86efa234a56019f5727a86/Library/MacTeXRequirement.rb), although I don’t believe this would be suitable for homebrew-core. It seems like #83738 would need to be merged before LilyPond can be included in homebrew-core.

#83738 is linux-only, and this formula is mac-only.

Reply:
> You should make formula for these dependencies if you need them.

Is it possible to build these dependencies directly in this formula? Tex Live is very big and already have a cask (mactex and mactex-no-gui)

Reply:
@jsfelix:

> #83738 is linux-only, and this formula is mac-only.

Not necessarily:

* https://github.com/Homebrew/homebrew-core/pull/83738#issuecomment-906190278
* https://github.com/Homebrew/homebrew-core/pull/83738#issuecomment-906195943
* https://github.com/Homebrew/homebrew-core/pull/83738#issuecomment-917685110

Reply:
Ok, so I can wait for #83738 and use texlive formula as a dependency.

Reply:
> I should also note that according to LilyPond’s documentation, using Guile v2 with LilyPond is not officially supported:
> 
> https://lilypond.org/doc/v2.22/Documentation/contributor/requirements-for-running-lilypond

I am use lilypond built with guile 2 for many years in macOS and I never had problems. Maybe we can keep guile2, or build guile 1.8 inside, because it is a very old and deprecated version.